src/jit/hashbv.cpp

```diff
@@ -1964,58 +1964,51 @@ indexType hashBvIterator::nextBit()
     // printf("in nextbit for bv:\n");
     // this->bv->dump();
 
-    if (!this->currNode)
+    if (this->currNode == nullptr)
     {
         this->nextNode();
     }
 
-top:
-
-    if (!this->currNode)
+    while (this->currNode != nullptr)
     {
-        return NOMOREBITS;
-    }
-
-more_data:
-    if (!this->current_data)
-    {
-        current_element++;
-        // printf("current element is %d\n", current_element);
-        // reached the end of this node
-        if (current_element == (indexType) this->currNode->numElements())
-        {
-            // printf("going to next node\n");
-            this->nextNode();
-            goto top;
-        }
-        else
+        for (;;)
         {
-            assert(current_element < (indexType) this->currNode->numElements());
-            // printf("getting more data\n");
-            current_data = this->currNode->elements[current_element];
-            current_base = this->currNode->baseIndex + current_element * BITS_PER_ELEMENT;
-            goto more_data;
-        }
-    }
-    else
-    {
-        while (current_data)
-        {
-            if (current_data & 1)
+            if (this->current_data == 0)
             {
-                current_data >>= 1;
-                current_base++;
-
-                return current_base - 1;
+                current_element++;
+                // printf("current element is %d\n", current_element);
+                // reached the end of this node
+                if (current_element == (indexType) this->currNode->numElements())
+                {
+                    // printf("going to next node\n");
+                    this->nextNode();
+                    break;   // break inner loop
+                }
+                else
+                {
+                    assert(current_element < (indexType) this->currNode->numElements());
+                    // printf("getting more data\n");
+                    current_data = this->currNode->elements[current_element];
+                    current_base = this->currNode->baseIndex + current_element * BITS_PER_ELEMENT;
+                }
             }
             else
             {
-                current_data >>= 1;
-                current_base++;
+                while (current_data != 0)
+                {
+                    current_data >>= 1;
+                    current_base++;
+
+                    if ((current_data & 1) != 0)
```

Comment:
I think that this refactoring is incorrect. Previously, this test was done before right-shifting `current_data`, but now it is done after the shift.